### PR TITLE
Migrate from MQCs in persisted validation data to merkle proofs

### PR DESCRIPTION
**Pull request by pepyakin**
- [x] Run this locally to check if it works

### CODE REVIEW

**Comment by pepyakin** [comment]
Some changes related to tests were lost somehow. WIP

**Comment by pepyakin** [comment]
I cannot test this locally, blocked by https://github.com/paritytech/polkadot/issues/2373

I think we can still review and land this and test the whole thing asynchronously

File: test/relay-sproof-builder/src/lib.rs [hunk 1]
```diff
@@ -44,14 +55,43 @@ impl Default for RelayStateSproofBuilder {
...
+	///
+	/// If there is no channel, a new default one is created.
+	///
+	/// It also updates the `hrmp_ingress_channel_index`, creating it if needed.
+	pub fn upsert_inbound_channel(&mut self, sender: ParaId) -> &mut AbridgedHrmpChannel {
+		let in_index = self.hrmp_ingress_channel_index.get_or_insert_with(Vec::new);
+		match in_index.binary_search(&sender) {
+			Ok(_idx) => {}
+			Err(idx) => in_index.insert(idx, sender),
+		}
```
Comment (bkchr) [inline test/relay-sproof-builder/src/lib.rs:78]:
```suggestion
		if let Err(idx) = in_index.binary_search(&sender) {
			in_index.insert(idx, sender);
		}
```

File: test/relay-sproof-builder/src/lib.rs [hunk 2]
```diff
@@ -44,14 +55,43 @@ impl Default for RelayStateSproofBuilder {
...
 	}
 }
 
 impl RelayStateSproofBuilder {
+	/// Returns a mutable reference to HRMP channel metadata for a channel (`sender`, `self.para_id`).
+	///
+	/// If there is no channel, a new default one is created.
+	///
+	/// It also updates the `hrmp_ingress_channel_index`, creating it if needed.
+	pub fn upsert_inbound_channel(&mut self, sender: ParaId) -> &mut AbridgedHrmpChannel {
```
Comment (bkchr) [inline test/relay-sproof-builder/src/lib.rs:73]: `upsert` I needed to google this :D But I also got no better idea.

Comment (pepyakin) [reply test/relay-sproof-builder/src/lib.rs:73]: Rust uses `get_or_insert`, [here](https://doc.rust-lang.org/1.48.0/core/option/enum.Option.html#method.get_or_insert). Perhaps we could use it, but that seemed to me a bit overly long.

Comment (bkchr) [reply test/relay-sproof-builder/src/lib.rs:73]: Yeah IDK

Comment (bkchr) [reply test/relay-sproof-builder/src/lib.rs:73]: I also thought about this, but it would probably too long as you said.

**Pull request merged**